### PR TITLE
Add and use chatAgents variable in Delegating Chat Agent

### PR DESCRIPTION
**Pull request by planger**
#### What it does

* Add \chatAgents that only includes actual chat agents not all agents
* Use \chatAgents instead of \agents in Delegating Chat Agent to avoid delegating to non-chat agents

#### How to test

Test that delegating chat agent never delegates to a non-chat agent (e.g. terminal agent).
Also you can inspect the system prompt after it has been resolved with variables in the delegating chat agent (see delegating-chat-agent.ts:89).

#### Follow-ups

N/A

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-chat/src/common/chat-agents-variable-contribution.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    resolveAgentsVariable(_request: AIVariableResolutionRequest): ResolvedChatAgentsVariable {
+        const agents = this.agents.getAgents().map(agent => ({
+            id: agent.id,
+            name: agent.name,
+            description: agent.description
+        }));
+        return { variable: CHAT_AGENTS_VARIABLE, agents, value: JSON.stringify(agents) };
```
Comment (sdirix) [inline packages/ai-chat/src/common/chat-agents-variable-contribution.ts:74]: Nitpick: Stringify works but the prompt looks very ugly then, with array notation, quotation marks etc. Personally I would "nicely" format this into a nested bulleted list or paragraphs.

**Review by sdirix — APPROVED** [review]
Works for me

File: packages/ai-chat/src/common/chat-agents-variable-contribution.ts (hunk 1 above)
Comment (planger) [reply packages/ai-chat/src/common/chat-agents-variable-contribution.ts:74]: Sure, good idea! Maybe this also improves the LLM processing. Let me append a small commit in a sec.

Comment (planger) [reply packages/ai-chat/src/common/chat-agents-variable-contribution.ts]: WDYT about 904eb64

**Review by sdirix — APPROVED** [review]

**Pull request merged**